src/NUnitFramework/framework/Attributes/RandomAttribute.cs

```diff
@@ -176,7 +178,7 @@ public IEnumerable GetData(IParameterInfo parameter)
                     _source = new SByteDataSource(_count);
                 else if (parmType == typeof(decimal))
                     _source = new DecimalDataSource(_count);
-                else if (parmType.IsEnum)
+                else if (parmType.GetTypeInfo().IsEnum)
```

Comment:
The majority of the changes are of this nature. Many of the properties and methods that we used to find directly on type are now accessed through the TypeInfo. On older platforms, I created an extension method for this on Type that simply returns the type.